pkg/build/builder/docker.go

```diff
@@ -69,10 +70,19 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, d.urlTimeout, os.Stdin, d.gitClient)
 	if err != nil {
+		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
+		d.build.Status.Message = StatusMessageFetchSourceFailed
+		if updateErr := updateStatusRetrier(d.build, d.client); updateErr != nil {
```

Comment:
I promised clayton we would not introduce additional build object updates w/ this feature.  Instead of setting reason/message here, returning an error, and then having the caller update the build status to "failed", can you find a way to return the reason/message values to the caller and have the caller set those fields when it is setting the build status to failed?

(this comment applies to all the locations where you are just setting reason/message)